elastic/kibana
# [Search Profiler] Migrate server to new es-js client

Part of https://github.com/elastic/kibana/issues/73973

This PR migrates Search Profiler to use the new ES JS client and adds API integration tests.

### How to test:
Verify there have been no regressions to Search Profiler.

1. Navigate to Search Profiler in Dev Tools.
2. Verify the default configuration yields results in the "Query Profile" panel.
3. Verify an invalid configuration (e.g., specifying an index that does not exist) shows an error notification and no results in the "Query Profile" panel.
